src/Api/Vault/Models/Response/CipherResponseModel.cs

```diff
@@ -96,26 +98,41 @@ public CipherMiniResponseModel(Cipher cipher, IGlobalSettings globalSettings, bo
 
 public class CipherResponseModel : CipherMiniResponseModel
 {
-    public CipherResponseModel(CipherDetails cipher, IGlobalSettings globalSettings, string obj = "cipher")
+    public CipherResponseModel(
+        CipherDetails cipher,
+        User user,
+        OrganizationAbility organizationAbility,
+        IGlobalSettings globalSettings,
+        string obj = "cipher")
         : base(cipher, globalSettings, cipher.OrganizationUseTotp, obj)
     {
         FolderId = cipher.FolderId;
         Favorite = cipher.Favorite;
         Edit = cipher.Edit;
         ViewPassword = cipher.ViewPassword;
+        Permissions = new CipherPermissionsResponseModel
+        {
+            Delete = NormalCipherPermissions.CanDelete(user, cipher, organizationAbility),
+            Restore = NormalCipherPermissions.CanRestore(user, cipher, organizationAbility)
```

Comment:
Suggestion: this is probably better encapsulated in the ctor for the `CipherPermissionsResponseModel`.